### PR TITLE
Configure .NET analyzer and code style settings

### PR DESCRIPTION
**Pull request by Marioalexsan**
## Description

This PR switches code style setup from `.editorconfig` + `omnisharp.json` to `.editorconfig` + .NET's built-in analyzers. The main idea is to setup proper code rules that can also be enforced at compile time if needed.

I've split it into two commits: the one that configures the rules, and the one that fixes all of the warnings generated by said rules.

## How to test?

Grab the branch and build it, then go back a commit and check what warnings were being generated for the previous version of the code base.

## Breaking Changes

- Interfaces were changed to use the `I`-prefix in their name
- Some function parameters name were changed to `camelCase` instead of `PascalCase`
- DllImport names for SFML where changed to use a capital first letter, e.g. `SFML.Audio` instead of `SFML.audio`
- Transform uses capitalized members (`M1` instead of `m1`) that are internally accessible

### CODE REVIEW

File: examples/sound_capture/sound_capture.csproj [hunk 1]
```diff
@@ -1,6 +1,8 @@
 <Project Sdk="Microsoft.NET.Sdk">
 
-	<PropertyGroup>
+  <Import Project="../../SFML.CodeStyle.props" />
+
+  <PropertyGroup>
 		<OutputType>Exe</OutputType>
 		<TargetFramework>net48</TargetFramework>
 		<RootNamespace>sound_capture</RootNamespace>
```
Comment (eXpl0it3r) [inline examples/sound_capture/sound_capture.csproj:8]: Can you reformat all *.csproj to use spaces?

File: examples/visualbasic/OpenGL.vb [hunk 2]
```diff
@@ -7,37 +6,38 @@ Imports OpenTK
...
+        }
 
         ' Create main window
         window = New RenderWindow(New VideoMode(800, 600), "SFML graphics with OpenGL (Visual Basic)", Styles.Default, contextSettings)
         window.SetVerticalSyncEnabled(True)
 
         ' Initialize OpenTK
-        Toolkit.Init()
-        Dim context As GraphicsContext = New GraphicsContext(New ContextHandle(IntPtr.Zero), Nothing)
+        Dim unused = Toolkit.Init()
```
Comment (eXpl0it3r) [inline examples/visualbasic/OpenGL.vb:27]: maybe it can be `unused1` and alter `unused2` instead of `unused` and `unused1`

Comment (Marioalexsan) [reply examples/visualbasic/OpenGL.vb]: Changed them to 1 & 2.

File: examples/netcore/Program.cs [hunk 3]
```diff
@@ -36,8 +36,10 @@ private static SoundBuffer GenerateSineWave(double frequency, double volume, int
             uint sampleRate = 44100;
             var samples = new short[seconds * sampleRate];
 
-            for (int i = 0; i < samples.Length; i++)
-                samples[i] = (short)(Math.Sin(frequency * (2 * Math.PI) * i / sampleRate) * volume * short.MaxValue);
+            for (var i = 0; i < samples.Length; i++)
+            {
+                samples[i] = (short)( Math.Sin(frequency * ( 2 * Math.PI ) * i / sampleRate) * volume * short.MaxValue );
```
[the remark this answers is not in the document]
Comment (Marioalexsan) [reply examples/netcore/Program.cs]: I've pushed a commit with the spacing removed.

File: examples/netcore/Program.cs [hunk 4]
```diff
@@ -1,14 +1,14 @@
...
 namespace netcore
 {
-    class Program
+    internal class Program
     {
-        static void Main(string[] args)
+        private static void Main()
         {
             var shape = new RectangleShape(new Vector2f(100, 100))
             {
```
[the remark this answers is not in the document]
Comment (Marioalexsan) [reply examples/netcore/Program.cs:14]: Done.

File: examples/visualbasic/My Project/AssemblyInfo.vb [hunk 5]
```diff
@@ -8,17 +7,17 @@ Imports System.Runtime.InteropServices
...
-<Assembly: AssemblyDescription("")> 
-<Assembly: AssemblyCompany("Wam")> 
-<Assembly: AssemblyProduct("visualbasic")> 
-<Assembly: AssemblyCopyright("Copyright © Wam 2008")> 
-<Assembly: AssemblyTrademark("")> 
+<Assembly: AssemblyTitle("visualbasic")>
+<Assembly: AssemblyDescription("")>
+<Assembly: AssemblyCompany("Wam")>
+<Assembly: AssemblyProduct("visualbasic")>
+<Assembly: AssemblyCopyright("Copyright © Wam 2008")>
```
[the remark this answers is not in the document]
Comment (Marioalexsan) [reply examples/visualbasic/My Project/AssemblyInfo.vb]: I've changed the info a bit, although I'm not sure if we even need AssemblyInfo files, given that `.csproj` should have support for these fields.

[the remark this answers is not in the document]
Comment (eXpl0it3r) [reply examples/visualbasic/My Project/AssemblyInfo.vb]: No idea if VB got similar features are C#

File: src/SFML.Graphics/RenderTexture.cs [hunk 6]
```diff
@@ -266,58 +233,43 @@ public Vector2i MapCoordsToPixel(Vector2f point, View view)
...
         ////////////////////////////////////////////////////////////
         /// <summary>
         /// Target texture of the render texture
         /// </summary>
         ////////////////////////////////////////////////////////////
-        public Texture Texture
-        {
-            get { return myTexture; }
-        }
+        public Texture Texture { get; } = null;
```
Comment (eXpl0it3r) [inline src/SFML.Graphics/RenderTexture.cs:265]: Is the `null` init not unnecessary as it's the default value?

File: src/SFML.Graphics/Shape.cs [hunk 7]
```diff
@@ -194,98 +189,86 @@ public Shape(Shape copy) :
...
+        private delegate UIntPtr GetPointCountCallbackType(IntPtr userData);
 
         [UnmanagedFunctionPointer(CallingConvention.Cdecl)]
-        private delegate Vector2f GetPointCallbackType(UIntPtr index, IntPtr UserData);
+        private delegate Vector2f GetPointCallbackType(UIntPtr index, IntPtr userData);
 
         private readonly GetPointCountCallbackType myGetPointCountCallback;
         private readonly GetPointCallbackType myGetPointCallback;
 
         private Texture myTexture = null;
```
Comment (eXpl0it3r) [inline src/SFML.Graphics/Shape.cs:225]: Can we add a rule to use `_`-prefix instead of `my`?

Comment (Marioalexsan) [reply src/SFML.Graphics/Shape.cs]: Done.

File: src/SFML.Graphics/RenderTexture.cs (hunk 6 above)
Comment (Marioalexsan) [reply src/SFML.Graphics/RenderTexture.cs]: It's not, I've configured a rule to warn about it in the future.

**Review by eXpl0it3r — APPROVED** [review]
Looks good!

You may want to squash merge, unless you think the separate commits help in some way

**Pull request merged**